### PR TITLE
Spring integration

### PR DESCRIPTION
**Pull request by pizzaeueu**

### CODE REVIEW

File: src/main/java/com/inquirer/servlets/LoginPageServlet.java [hunk 1]
```diff
@@ -16,14 +19,18 @@
 import javax.servlet.http.HttpSession;
 import java.io.IOException;
 
+@Controller
+@RequestMapping("login")
 public class LoginPageServlet extends HttpServlet{
```
Comment (mafitov) [inline src/main/java/com/inquirer/servlets/LoginPageServlet.java]: Remove HttpServlet and rename to LoginController

File: src/main/java/com/inquirer/servlets/LogOutServlet.java [hunk 2]
```diff
@@ -1,13 +1,20 @@
...
+
 import javax.servlet.http.HttpServlet;
 import javax.servlet.http.HttpServletRequest;
 import javax.servlet.http.HttpServletResponse;
 import javax.servlet.http.HttpSession;
 import java.io.IOException;
 
+@Controller
+@RequestMapping("logout")
 public class LogOutServlet extends HttpServlet {
```
Comment (mafitov) [inline src/main/java/com/inquirer/servlets/LogOutServlet.java]: Remove HttpServlet and rename to LogoutController

File: src/main/java/com/inquirer/dao/impl/UserDaoImpl.java [hunk 3]
```diff
@@ -4,11 +4,13 @@
...
 import com.inquirer.utils.DaoHelper;
 import org.apache.log4j.Logger;
+import org.springframework.stereotype.Repository;
 
 import java.sql.*;
 import java.util.ArrayList;
 import java.util.List;
 
+@Repository
 public class UserDaoImpl implements UserDao {
```
Comment (mafitov) [inline src/main/java/com/inquirer/dao/impl/UserDaoImpl.java]: Rename to UserRepository

File: src/main/java/com/inquirer/dao/impl/QuestionDaoImpl.java [hunk 4]
```diff
@@ -4,11 +4,13 @@
...
 import com.inquirer.utils.DaoHelper;
 import org.apache.log4j.Logger;
+import org.springframework.stereotype.Repository;
 
 import java.sql.*;
 import java.util.ArrayList;
 import java.util.List;
 
+@Repository
 public class QuestionDaoImpl implements QuestionDao {
```
Comment (mafitov) [inline src/main/java/com/inquirer/dao/impl/QuestionDaoImpl.java]: Rename to QuestionRepository

File: src/main/java/com/inquirer/dao/impl/AnswerDaoImpl.java [hunk 5]
```diff
@@ -6,13 +6,15 @@
...
+import org.springframework.stereotype.Repository;
 
 import java.sql.PreparedStatement;
 import java.sql.ResultSet;
 import java.sql.SQLException;
 import java.util.ArrayList;
 import java.util.List;
 
+@Repository
 public class AnswerDaoImpl implements AnswerDao {
```
Comment (mafitov) [inline src/main/java/com/inquirer/dao/impl/AnswerDaoImpl.java]: Rename to AnswerRepository

File: src/main/java/com/inquirer/servlets/ResultsPageServlet.java [hunk 6]
```diff
@@ -11,16 +11,22 @@
...
 
 import javax.servlet.http.HttpServlet;
 import javax.servlet.http.HttpServletRequest;
 import javax.servlet.http.HttpServletResponse;
 import javax.servlet.http.HttpSession;
 import java.io.IOException;
 
+@Controller
+@RequestMapping("results")
 public class ResultsPageServlet extends HttpServlet {
```
Comment (mafitov) [inline src/main/java/com/inquirer/servlets/ResultsPageServlet.java]: Remove HttpServlet and rename to ResultsController

File: src/main/java/com/inquirer/servlets/TestPageServlet.java [hunk 7]
```diff
@@ -21,6 +24,8 @@
 import java.util.ArrayList;
 import java.util.List;
 
+@Controller
+@RequestMapping("test")
 public class TestPageServlet extends HttpServlet {
```
Comment (mafitov) [inline src/main/java/com/inquirer/servlets/TestPageServlet.java]: Remove HttpServlet and rename to TestController

File: src/main/webapp/WEB-INF/dispatcher-servlet.xml [hunk 8]
```diff
@@ -0,0 +1,40 @@
...
+    http://www.springframework.org/schema/context/spring-context.xsd
+    http://www.springframework.org/schema/mvc
+         http://www.springframework.org/schema/mvc/spring-mvc-3.0.xsd">
+
+    <mvc:annotation-driven/>
+    <mvc:resources mapping="/css/**" location="/css/"
+                   cache-period="31556926"/>
+
+
+    <context:component-scan base-package="com.inquirer.servlets"/>
```
Comment (mafitov) [inline src/main/webapp/WEB-INF/dispatcher-servlet.xml]: base-package=com.inquirer

File: src/main/java/com/inquirer/servlets/LogOutServlet.java [hunk 9]
```diff
@@ -1,13 +1,20 @@
 package com.inquirer.servlets;
```
Comment (mafitov) [inline src/main/java/com/inquirer/servlets/LogOutServlet.java]: Rename to com.inquirer.controller

File: src/main/java/com/inquirer/services/impl/AnswerServiceImpl.java [hunk 10]
```diff
@@ -19,7 +21,7 @@
     private static final Logger LOGER = Logger.getLogger(AnswerServiceImpl.class);
 
     public AnswerServiceImpl(){
-        answerDao = new AnswerDaoImpl();
+        answerDao = new AnswerRepository();
```
Comment (viacheslav-bogdanov) [inline src/main/java/com/inquirer/services/impl/AnswerServiceImpl.java]: Use dependency injection

File: src/main/java/com/inquirer/services/impl/QuestionServiceImpl.java [hunk 11]
```diff
@@ -1,22 +1,24 @@
...
+@Service
 public class QuestionServiceImpl implements QuestionService {
 
     private static final Logger LOGER = Logger.getLogger(QuestionServiceImpl.class);
 
     private QuestionDao questionDao;
 
     public QuestionServiceImpl(){
-        questionDao = new QuestionDaoImpl();
+        questionDao = new QuestionRepository();
```
Comment (viacheslav-bogdanov) [inline src/main/java/com/inquirer/services/impl/QuestionServiceImpl.java]: See above

File: src/main/webapp/WEB-INF/applicationContext.xml [hunk 12]
```diff
@@ -0,0 +1,6 @@
+<?xml version="1.0" encoding="UTF-8"?>
```
Comment (viacheslav-bogdanov) [inline src/main/webapp/WEB-INF/applicationContext.xml]: Move applicationContext.xml to src/main/resource folder

File: src/main/java/com/inquirer/controller/AuthFilter.java [hunk 13]
```diff
@@ -1,6 +1,4 @@
-package com.inquirer.servlets;
-
-import com.inquirer.models.User;
+package com.inquirer.controller;
 
 import javax.servlet.*;
```
Comment (mafitov) [inline src/main/java/com/inquirer/controller/AuthFilter.java]: Avoid using "*" in import section

File: src/main/java/com/inquirer/controller/LogOutController.java [hunk 14]
```diff
@@ -0,0 +1,21 @@
...
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+import javax.servlet.http.HttpSession;
+import java.io.IOException;
+
+@Controller
+@RequestMapping("logout")
+public class LogOutController {
+    @RequestMapping(method = RequestMethod.GET)
+    protected String doGet(HttpServletRequest request, HttpServletResponse response) throws IOException {
```
Comment (mafitov) [inline src/main/java/com/inquirer/controller/LogOutController.java]: You can specify any name here. Not doGet or doPost.

File: src/main/java/com/inquirer/controller/LoginController.java [hunk 15]
```diff
@@ -0,0 +1,50 @@
...
+import javax.servlet.http.HttpSession;
+import java.io.IOException;
+
+@Controller
+@RequestMapping("login")
+public class LoginController {
+    @Autowired private UserService userLoginService;
+
+    @RequestMapping(method = RequestMethod.GET)
+    protected String doGet(HttpServletRequest request, HttpServletResponse response) throws IOException {
```
Comment (viacheslav-bogdanov) [inline src/main/java/com/inquirer/controller/LoginController.java]: Remove unnecessary parameteres

Comment (mafitov) [inline src/main/java/com/inquirer/controller/LoginController.java]: No need to inject HttpServletRequest and HttpServletResponse. Please fix it everywhere.

File: src/main/java/com/inquirer/controller/LoginController.java [hunk 16]
```diff
@@ -0,0 +1,50 @@
...
+    @RequestMapping(method = RequestMethod.POST)
+    public String doPost(HttpServletRequest request,HttpServletResponse response) throws IOException, ServletException {
+        User user = new User();
+
+        user.setName(request.getParameter("login"));
+
+        //UserServiceImpl userLoginService = new UserServiceImpl();
+
+        if(userLoginService.isUserExist(user)) {
+            HttpSession session = request.getSession();
```
Comment (mafitov) [inline src/main/java/com/inquirer/controller/LoginController.java]: HttpSession can be injected into method params by Spring, like HttpServletRequest or HttpServletResponse here.

File: src/main/java/com/inquirer/dao/impl/AnswerRepository.java [hunk 17]
```diff
@@ -6,19 +6,21 @@
...
 
-public class AnswerDaoImpl implements AnswerDao {
+@Repository
+public class AnswerRepository implements AnswerDao {
 
     private static final String SELECT_ANSWER_FOR_QUESTION_QUERY = "SELECT * FROM answer WHERE question = ?";
     private static final String SELECT_ANSWER_BY_ID_QUERY = "SELECT * FROM answer WHERE answer.id = ?";
 
-    private static final Logger LOGER = Logger.getLogger(AnswerDaoImpl.class);
+    private static final Logger LOGER = Logger.getLogger(AnswerRepository.class);
```
Comment (mafitov) [inline src/main/java/com/inquirer/dao/impl/AnswerRepository.java]: Should be AnswerRepositoryImpl

File: src/main/java/com/inquirer/controller/TestPageController.java [hunk 18]
```diff
@@ -53,12 +58,11 @@ protected void doGet(HttpServletRequest request, HttpServletResponse response) t
         request.setAttribute("questionTitle", questionTitle);
         request.setAttribute("answers", answers);
```
Comment (viacheslav-bogdanov) [inline src/main/java/com/inquirer/controller/TestPageController.java]: You can use this interface to put necessary attributes 
http://docs.spring.io/spring-framework/docs/4.3.x/javadoc-api/org/springframework/ui/Model.html

File: src/main/java/com/inquirer/dao/impl/AnswerRepository.java [hunk 19]
```diff
@@ -6,19 +6,21 @@
...
 
 import java.sql.PreparedStatement;
 import java.sql.ResultSet;
 import java.sql.SQLException;
 import java.util.ArrayList;
 import java.util.List;
 
-public class AnswerDaoImpl implements AnswerDao {
+@Repository
+public class AnswerRepository implements AnswerDao {
```
Comment (mafitov) [inline src/main/java/com/inquirer/dao/impl/AnswerRepository.java]: Should be "public class AnswerRepositoryImpl implements AnswerRepository"

File: src/main/webapp/WEB-INF/web.xml [hunk 20]
```diff
@@ -5,12 +5,42 @@
...
+    <session-config>
+        <session-timeout>
+            30
+        </session-timeout>
+    </session-config>
+    <!-- Spring -->
+
+    <!-- OLD -->
+    <!--
     <servlet>
```
Comment (mafitov) [inline src/main/webapp/WEB-INF/web.xml]: Remove servlet section

File: src/main/java/com/inquirer/controller/LogOutController.java [hunk 21]
```diff
@@ -0,0 +1,20 @@
...
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+import javax.servlet.http.HttpSession;
+import java.io.IOException;
+
+@Controller
+@RequestMapping("logout")
+public class LogOutController {
+    @RequestMapping(method = RequestMethod.GET)
+    protected String logout(HttpSession session) throws IOException {
```
Comment (mafitov) [inline src/main/java/com/inquirer/controller/LogOutController.java]: Do we need throws IOException ?

File: src/main/java/com/inquirer/controller/LoginController.java [hunk 22]
```diff
@@ -0,0 +1,48 @@
...
+import javax.servlet.http.HttpSession;
+import java.io.IOException;
+
+@Controller
+@RequestMapping("login")
+public class LoginController {
+    @Autowired private UserService userLoginService;
+
+    @RequestMapping(method = RequestMethod.GET)
+    protected String login() throws IOException {
```
Comment (mafitov) [inline src/main/java/com/inquirer/controller/LoginController.java]: throws IOException

File: src/main/java/com/inquirer/controller/LoginController.java [hunk 23]
```diff
@@ -0,0 +1,48 @@
...
+        return "inquirer.loginPage";
+    }
+    @RequestMapping(method = RequestMethod.POST)
+    public String acceptUser(HttpServletRequest request, HttpSession session, Model model) throws IOException, ServletException {
```
Comment (mafitov) [inline src/main/java/com/inquirer/controller/LoginController.java]: throws IOException, ServletException

File: src/main/java/com/inquirer/controller/ResultsPageController.java [hunk 24]
```diff
@@ -1,31 +1,31 @@
...
+@Controller
+@RequestMapping("results")
+public class ResultsPageController {
 
-    @Override
-    protected void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException {
+    @Autowired private ResultService resultService;
+    @RequestMapping(method = RequestMethod.GET)
+    protected String watchResult(HttpServletRequest request,HttpSession session, Model model) throws IOException {
         User user = (User) request.getSession().getAttribute("user");
```
Comment (mafitov) [inline src/main/java/com/inquirer/controller/ResultsPageController.java]: request.getSession() -> session

File: src/main/java/com/inquirer/controller/ResultsPageController.java [hunk 25]
```diff
@@ -1,31 +1,31 @@
...
-public class ResultsPageServlet extends HttpServlet {
+@Controller
+@RequestMapping("results")
+public class ResultsPageController {
 
-    @Override
-    protected void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException {
+    @Autowired private ResultService resultService;
+    @RequestMapping(method = RequestMethod.GET)
+    protected String watchResult(HttpServletRequest request,HttpSession session, Model model) throws IOException {
```
Comment (mafitov) [inline src/main/java/com/inquirer/controller/ResultsPageController.java]: Remove HttpServletRequest request

File: src/main/java/com/inquirer/controller/LoginController.java (hunk 23 above)
Comment (mafitov) [inline src/main/java/com/inquirer/controller/LoginController.java]: Remove HttpServletRequest request

File: src/main/java/com/inquirer/controller/TestPageController.java [hunk 26]
```diff
@@ -0,0 +1,78 @@
...
+        model.addAttribute("questionTitle", questionTitle);
+        model.addAttribute("answers", answers);
+
+        return "inquirer.testPage";
+    }
+
+    @RequestMapping(method = RequestMethod.POST)
+    public String acceptQuestion(HttpServletRequest request, HttpSession session) throws IOException, ServletException {
+
+        if (request.getParameter("userAnswerInput")!=null) {
```
Comment (mafitov) [inline src/main/java/com/inquirer/controller/TestPageController.java]: Use @RequestParam

File: src/main/webapp/WEB-INF/web.xml [hunk 27]
```diff
@@ -5,49 +5,40 @@
...
+        <session-timeout>
+            30
+        </session-timeout>
+    </session-config>
+    <!-- Spring -->
+
+       <filter>
         <filter-name>AuthenticationFilter</filter-name>
-        <filter-class>com.inquirer.servlets.AuthFilter</filter-class>
+        <filter-class>com.inquirer.controller.AuthFilter</filter-class>
```
Comment (mafitov) [inline src/main/webapp/WEB-INF/web.xml]: Should be com.inquirer.filter package

File: src/main/java/com/inquirer/controller/TestPageController.java [hunk 28]
```diff
@@ -0,0 +1,78 @@
...
+    @Autowired private QuestionServiceImpl questionService;
+
+    @RequestMapping(method = RequestMethod.GET)
+    protected String getQuestion(HttpServletRequest request, Model model) throws IOException {
+
+        Question currentQuestion = questionService.getQuestionByNumber(Integer.parseInt(request.getParameter("question")));
+        String questionTitle = currentQuestion.getTitle();
+
+        List<Answer> answers = new ArrayList<>();
+        try {
```
Comment (mafitov) [inline src/main/java/com/inquirer/controller/TestPageController.java]: Move try/catch to service layer

**Comment by mafitov** [comment]
:+1:

**Pull request merged**